File: signature/mod_perl.go
package signature

var ModPerlSignature = Signature{
	Name:        "mod_perl",
	Description: "mod_perl brings together the full power of the Perl programming language and the Apache HTTP server.",
	Cpe:         "cpe:/a:apache:mod_perl",

	DetectPattern: DetectPattern{
		Headers: []Header{{
			Name:  "server",
			Value: "mod_perl",
		}},
	},
	VersionPattern: VersionPattern{
		Headers: []Header{{
			Name:  "server",
			Value: `mod_perl/(\d+\.\d+\.\d+)`,
		}},
	},

	OnlySameHost: true,
}

File: src/signatures/technologies/mod_perl.ts
import type { Signature } from "../_types.js";
import { apacheHttpServerSignature } from "./apache.js";
import { perlSignature } from "./perl.js";

export const modPerlSignature: Signature = {
  name: "mod_perl",
  description:
    "Mod_perl is an optional module for the Apache HTTP server. It embeds a Perl interpreter into the Apache server.",
  cpe: "cpe:2.3:a:apache:mod_perl:*:*:*:*:*:*:*:*",
  rule: {
    confidence: "high",
    headers: {
      Server: "mod_perl(?:/([\\d\\.]+))?",
    },
  },
  impliedSoftwares: [perlSignature.name, apacheHttpServerSignature.name],
};
